### PR TITLE
Docker containers refactor

### PR DESCRIPTION
**Pull request by karun-singh**
- Moved Dockerfile and context to docker/ directory for all associated component 
- [x] Remove dockerfiles from previous locations

### CODE REVIEW

File: docker/redis/redis-autoscaler/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+RUN chmod g+rwX /opt/bitnami
+
+COPY rootfs /
+RUN /opt/bitnami/scripts/redis-cluster/postunpack.sh
+ENV APP_VERSION="7.0.2" \
+    BITNAMI_IMAGE_VERSION="7.0.2-debian-11-r8" \
+    BITNAMI_APP_NAME="redis-cluster" \
+    PATH="/opt/bitnami/redis/bin:/opt/bitnami/common/bin:$PATH"
+
+USER 0
```
Comment (abhi4578) [inline docker/redis/redis-autoscaler/Dockerfile:31]: Any reason the user 0  is used?

Comment (abhi4578) [reply docker/redis/redis-autoscaler/Dockerfile]: Needs to be tested with non-root user. resolving for now

**Comment by abhi4578** [comment]
Can i merge @karun-singh ?

**Comment by karun-singh** [comment]
> Can i merge @karun-singh ?

Yes, @abhi4578, looks good.

**Review by abhi4578 — APPROVED** [review]

**Pull request merged**